RecoVertex/PrimaryVertexProducer/src/DAClusterizerInZ_vect.cc

```diff
@@ -95,10 +95,19 @@ DAClusterizerInZ_vect::DAClusterizerInZ_vect(const edm::ParameterSet& conf) {
 namespace {
   inline double local_exp(double const& inp) { return vdt::fast_exp(inp); }
 
+  inline void local_exp_list(double const* __restrict__ arg_inp,
+                             double* __restrict__ arg_out,
+                             const unsigned arg_arr_size) {
+#pragma GCC ivdep
+    for (unsigned i = 0; i != arg_arr_size; ++i)
+      arg_out[i] = vdt::fast_exp(arg_inp[i]);
+  }
```

Comment:
this is the same as ```local_exp_list_range(arg_inp, arg_out, 0, arg_arr_size)```
I don't see an obvious a need for a separate implementation of this case.
Does it somehow vectorize differently?
If there is no difference, perhaps a synced-up update in the InZT_vect is in order as well

Reply:
The loop in local_exp_list_range will not vectorize even with the restrict attribute.

Reply:
ehm, OK; not even with gcc10?
I guess the ``` #pragma GCC ivdep``` remains there as just wishful thinking.
Perhaps a comment there in the code, like ```// did not auto-vectorize as of PR #30437```  would be useful for posterity.

Reply:
The loop in ```local_exp_list``` does vectorized because at least the starting index is fixed. This is not the case for ```local_exp_list_range```.

Reply:
I will go back and remove the pragma's where they have no effect.